contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -1328,6 +1339,126 @@ pltsql_post_parse_analyze(ParseState *pstate, Query *query, JumbleState *jstate)
 	}
 }
 
+static void
+checkForJsonAuto(Query *query)
+{
+	// Detect JSON AUTO
+	List* target = query->targetList;
+	bool isJsonAuto = false;
+	List* rtable;
+	RangeTblEntry* rte;
+	Query* subq;
+	List* subqRtable;
+	RangeTblEntry* subqRte;
+	RangeTblEntry* queryRte;
+	Alias *colnameAlias;
+	int numTables = 0;
+	int currTables = 0;
+	int currMax = 0;
+	forjson_table **tableInfoArr;
+	if(target) {
+		ListCell* lc = (ListCell*) target->elements;
+		if(lc && ((Node*) lc->ptr_value)->type == T_TargetEntry) {
+			TargetEntry* te = (TargetEntry*) lc->ptr_value;
+			if(te && strncmp(te->resname, "json", 4) == 0 && te->expr != NULL && ((Expr*) te->expr)->type == T_FuncExpr) {
+				List* args = ((FuncExpr*) te->expr)->args;
+				if(args != NULL && ((Node*) ((ListCell*) args->elements)->ptr_value)->type == T_Aggref) {
```

Comment:
Did you mean need to get the first elements of List arg here ?
I think better to write lfirst or proper function from pg_list.h